### DISCUSSION TITLE
Need a Source Generator post mortem post / doc / tutorial

### DISCUSSION THREAD

**Discussion by MithrilMan** [dotnet/runtime#61555] [Ideas]
From a dev PoV potentially interested in implementing open source libraries that leverage source generators, I feel the need of proper documentation/best practice/tutorial to introduce me into the Roslyn world of analyzers and source generator.

Current documentation is lackluster, on the web we can find some source generator source code but often is confused, sub-optimal implemented, based on old stuff and more broadly there is a lack of knowledge sharing about the topic.

In the hope of having access to more specific and in-depth documentation, could I suggest to publish a post-mortem blog post about some of your source generator implementation where you explain why it was implemented in a specific way and what are the best practice your are following while writing a source generator?

Thanks

**Comment by huoyaoyuan** [comment]
Source generator itself is still evolving. Any sample for Roslyn 3.8/3.9 is now outdated for 4.0. I think it's better to wait for source generator itself being mature. Currently there's still items in todo list, like attribute parsing.
